common/utils.go

```diff
@@ -58,40 +59,87 @@ func FindNetDevs(processID int, match func(string) bool) ([]NetDev, error) {
 	defer ns.Close()
 
 	err = WithNetNS(ns, func() error {
-		links, err := netlink.LinkList()
-		if err != nil {
-			return err
-		}
-		for _, link := range links {
-			if match(link.Attrs().Name) {
-				addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
+		return forEachLink(func(link netlink.Link) error {
+			if match(link) {
+				netDev, err := linkToNetDev(link)
 				if err != nil {
 					return err
 				}
-
-				netDev := NetDev{MAC: link.Attrs().HardwareAddr}
-				for _, addr := range addrs {
-					netDev.CIDRs = append(netDev.CIDRs, addr.IPNet)
-				}
-				netDevs = append(netDevs, netDev)
+				netDevs = append(netDevs, *netDev)
 			}
-		}
-		return nil
+			return nil
+		})
 	})
 
 	return netDevs, err
 }
 
+func forEachLink(f func(netlink.Link) error) error {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return err
+	}
+	for _, link := range links {
+		if err := f(link); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func linkToNetDev(link netlink.Link) (*NetDev, error) {
+	addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
+	if err != nil {
+		return nil, err
+	}
+
+	netDev := &NetDev{MAC: link.Attrs().HardwareAddr}
+	for _, addr := range addrs {
+		netDev.CIDRs = append(netDev.CIDRs, addr.IPNet)
+	}
+	return netDev, nil
+}
+
 // Lookup the weave interface of a container
 func GetWeaveNetDevs(processID int) ([]NetDev, error) {
-	return FindNetDevs(processID, func(name string) bool {
-		return strings.HasPrefix(name, "ethwe")
+	// Bail out if this container is running in the root namespace
+	nsToplevel, _ := netns.GetFromPid(1)
+	nsContainr, _ := netns.GetFromPid(processID)
+	if nsToplevel.Equal(nsContainr) {
```

Comment:
seems dangerous not to check for errors above. Is it really impossible for `netns.GetFromPid(1)` to fail?